### PR TITLE
Combat robots can be revived without defib

### PR DESCRIPTION
**Pull request by EvilMelonTheSecond**
## About The Pull Request
Makes combat robots revivable by clicking them with an empty hand. The revive damage threshold is lower than defibbing depending on your engi skill.

## Why It's Good For The Game
Engineers can now repair AND revive combat robots. Everybody else can do so but the revive threshold is 50*engiskill so a marine or a medic has to fully repair them before attempting revive. Engineers can revive at 150 health and ST can revive at 200. They can still be defibbed like normal at 200 and revival by hand also requires armor to be taken off.

No more medics dragging a robot to engi before defibbing. Engineers are (almost) fully responsible for robot health and safety. 

## Changelog
:cl:
add: Robots can be revived (or restarted) by hand.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 1]
```diff
@@ -34,6 +34,88 @@
...
+				if(src.wear_suit)
+					to_chat(H, span_warning("You can't reach [src]'s maintenance panel through their armor."))
+					return FALSE
+
+				var/mob/dead/observer/G = src.get_ghost()
+				if(istype(G))
+					notify_ghost(G, "<font size=3>Someone is trying to revive your body. Return to it if you want to be resurrected!</font>", ghost_sound = 'sound/effects/adminhelp.ogg', enter_text = "Enter", enter_link = "reentercorpse=1", source = src, action = NOTIFY_JUMP)
+				else if(!src.client)
+					//We couldn't find a suitable ghost, this means the person is not returning
+					to_chat(H, span_warning("Maintenance panel of [src] is locked tight, this unit refuses to be restarted."))
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/human/human_attackhand.dm:50]:
```suggestion
					to_chat(H, span_warning("The maintenance panel of [src] is locked tight, this unit refuses to be restarted."))
```

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 2]
```diff
@@ -34,6 +34,88 @@
...
+				span_notice("You complete the restart sequence of [src] and it sparks back to life!"))
+				src.visible_message(span_danger("[src]'s joints suddenly start moving."))
+
+				if(src.wear_suit)
+					to_chat(H, span_warning("You can't reach [src]'s maintenance panel through their armor."))
+					return FALSE
+
+				if((HAS_TRAIT(src, TRAIT_UNDEFIBBABLE ) || src.suiciding))
+					H.visible_message(span_warning("[src]'s backup battery blinks weakly, too late to save this one."),
+					span_warning("[src]'s backup battery ran dry, it cannot be restarted"))
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/human/human_attackhand.dm:75]:
```suggestion
					span_warning("[src]'s backup battery ran dry, it cannot be restarted."))
```

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 3]
```diff
@@ -34,6 +34,88 @@
...
+					H.visible_message(span_warning("[icon2html(src, viewers(H))] [src] buzzes: Restart sequence failed. No personality program detected."))
+					return FALSE
+
+				if(src.health <= restart_damage_threshold)
+					H.visible_message(span_warning("[icon2html(src, viewers(H))] [src] buzzes: Restart sequence failed. Sustained damage critical, repair damage and try again."))
+					return FALSE
+
+				src.visible_message(span_notice("[icon2html(src, viewers(H))] [src] beeps: Restart sequence successful."))
+				src.set_stat(UNCONSCIOUS)
+				src.emote("gasp")
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/human/human_attackhand.dm:102]: Robots do not gasp, do they?

**Comment by bluezorua** [comment]
by hand seems a bit too powerful, maybe by multitool?

**Comment by Lumipharon** [comment]
Just make them reboot themselves terminator style, after a brief windup once they've been repaired enough.

**Comment by Three333Digit** [comment]
I agree with autodefib, it's cool and unique.

**Review by TiviPlus — COMMENTED** [review]
I said it in disc but make it on multitool

**Comment by EvilMelonTheSecond** [comment]
Will remake one after med rework. The current version gotta change anyways.

**Pull request closed without merging**